.github/workflows/rust.yml

```diff
@@ -131,11 +143,27 @@ jobs:
             os: windows-latest
             output: cargo-binstall.exe
             archive: zip
+          - target: x86_64-unknown-linux-musl
+            os: ubuntu-latest
+            output: cargo-binstall
+            archive: tgz
 
     steps:
     - uses: actions/checkout@v2
     - uses: FranzDiebold/github-env-vars-action@v1.2.1
 
+    - name: Configure toolchain
+      if: ${{ matrix.target == 'x86_64-unknown-linux-musl' }}
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: stable
+        target:  ${{ matrix.target }}
+        override: true
+
+    - name: Install musl-tools
+      if: ${{ matrix.target == 'x86_64-unknown-linux-musl' }}
+      run: sudo apt-get install -y musl-tools
+
```

Comment:
Why do we need this here? We don't need a rust build environment in this build job, do we?

Reply:
> Why do we need this here? We don't need a rust build environment in this build job, do we?

Some of the dependencies of `cargo-binstall` (I think it was `rustls`, which contains a fair amount of `C` code) requires `musl-gcc` to be present.

Reply:
Aha, sorry I thought you were referring to the step in `build`.

For job `test`, it is because that we don't have pre-built binary for `x86_64-unknown-linux-musl` target yet, so the code goes back to running `cargo build` and it also failed because `libssl-1.1` is not present.

I will remove this.

Reply:
Done!